NEW TEST(265708@main): [ wk2 ] fast/mediastream/video-srcObject-set-twice.html is a flaky ImageOnlyFailure.

youennf opened the pull request:
#### a8801edb72066c945f69d34863752f34cb56f2a9
<pre>
NEW TEST(265708@main): [ wk2 ] fast/mediastream/video-srcObject-set-twice.html is a flaky ImageOnlyFailure.
<a href="https://bugs.webkit.org/show_bug.cgi?id=261920">https://bugs.webkit.org/show_bug.cgi?id=261920</a>
rdar://115868374

Reviewed by Jean-Yves Avenard.

The flakiness comes from the fact that the canvas capture video frame is not always properly rendered.
While we should fix this, we can make the test no longer flaky by relying on the existing take snapshot test.
This keeps the current coverage of the test which is to check that setting twice srcObject will still allow rendering video.

* LayoutTests/fast/mediastream/video-srcObject-set-twice-expected.html: Removed.
* LayoutTests/fast/mediastream/video-srcObject-set-twice-expected.txt: Added.
* LayoutTests/fast/mediastream/video-srcObject-set-twice.html:
* LayoutTests/platform/wk2/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/268908@main">https://commits.webkit.org/268908@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/14e2f0c3d74464ae16e2eaa765093e39fbc8573a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20244 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20676 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21314 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22144 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18892 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23930 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20844 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/20335 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20463 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20369 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17602 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22994 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17537 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18405 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/24662 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18614 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18581 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22643 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19161 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16271 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18366 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/18241 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5043 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22707 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18986 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jyavenard commented on LayoutTests/fast/mediastream/video-srcObject-set-twice.html:
```diff
@@ -42,44 +47,35 @@
+    if (!isPixelAlmostGreen(i, 50, canvas, data))
         return false;
-    while (isPixelGreen(++i, 50, canvas, data)) { };
+    while (isPixelAlmostGreen(++i, 50, canvas, data)) { };
```
where is this defined?

jyavenard commented on LayoutTests/fast/mediastream/video-srcObject-set-twice.html:
```diff
@@ -22,6 +22,12 @@
    return pixel.r === 0 && pixel.g === 128 && pixel.b === 0;
 }
 
+function isPixelGreen(x, y, canvas, data)
```
the name isn't particularly suitable nor represent what it's doing, it's is isPixelAlmostGreen

youennf committed:
NEW TEST(265708@main): [ wk2 ] fast/mediastream/video-srcObject-set-twice.html is a flaky ImageOnlyFailure.
https://bugs.webkit.org/show_bug.cgi?id=261920
rdar://115868374

Reviewed by Jean-Yves Avenard.

The flakiness comes from the fact that the canvas capture video frame is not always properly rendered.
While we should fix this, we can make the test no longer flaky by relying on the existing take snapshot test.
This keeps the current coverage of the test which is to check that setting twice srcObject will still allow rendering video.

* LayoutTests/fast/mediastream/video-srcObject-set-twice-expected.html: Removed.
* LayoutTests/fast/mediastream/video-srcObject-set-twice-expected.txt: Added.
* LayoutTests/fast/mediastream/video-srcObject-set-twice.html:
* LayoutTests/platform/wk2/TestExpectations:

Canonical link: https://commits.webkit.org/268908@main
